webapp/src/plugin.jsx

```diff
@@ -10,10 +10,15 @@ import AttachCommentToIssueModal from 'components/modals/attach_comment_to_issue
 import PluginId from 'plugin_id';
 
 import reducers from './reducers';
-import {handleConnectChange, getConnected, handleInstanceStatusChange} from './actions';
+import {handleConnectChange, getConnected, handleInstanceStatusChange, getSettings} from './actions';
 
 export default class Plugin {
     async initialize(registry, store) {
+        const settings = await getSettings(store.getState);
```

Comment:
Initialize blocks the webapp from loading until it's complete. Not sure it's a great idea to make a blocking request here. Maybe we can move the logic of visibility to the components?

Reply:
I thought that too, but here's my thinking:
1. There is an await already just a couple lines below, so we're already blocking the initialization.
1. It should be an extremely quick request.
1. Even if it blocks for 10 seconds, nothing blocks on the webapp, you just don't get the dropdown components until the request completes. (I slept the request on the server for 10 seconds to test it.) 

With that, I thought this was by far the simplest way of solving the problem. Yes, I could put another boolean in the state and do a test before returning the dropdown components (just like we do for instanceInstalled or userConnected), but then I'd have to add all that reducer code and selector code, and that's a lot of code for something simple. 

But, I'm open to doing it that way if you're set against this approach. :)

Reply:
@crspeller ^^

Reply:
If it only blocks the menus from appearing then it's fine. I thought it would block the whole webapp from loading. If it becomes an issue we can do the more complicated method.

Reply:
Nope, 100% doesn't block the mattermost webapp.